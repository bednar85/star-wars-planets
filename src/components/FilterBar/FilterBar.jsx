import React from 'react';
import PropTypes from 'prop-types';

function FilterBar(props) {
  const { filterFormHandler, searchFormHandler } = props;

  const renderInputs = inputData => {
    const { category, defaultChecked, labels, type } = inputData;

    return labels.map((label, index) => {
      const isCheckedByDefault =
        defaultChecked !== undefined ? defaultChecked === index : false;
      const labelKebabCased = label
        .toLowerCase()
        .replace(/[^a-zA-Z']/gi, ' ')
        .trim()
        .replace(/\s+/g, '-');

      return (
        <div
          key={`filter-form-input-${index}`}
          className="filter-form-input-wrapper"
        >
          <input
            type={type}
            className="filter-form-input"
            id={`${category}-${labelKebabCased}`}
            name={category}
            value={label}
            defaultChecked={isCheckedByDefault}
          />
          <label
            className="filter-form-label"
            htmlFor={`${category}-${labelKebabCased}`}
          >
            {label}
          </label>
        </div>
      );
    });
  };

  return (
    <div className="search-and-filter-forms">
      <form className="search-form" onChange={searchFormHandler}>
        <input type="text" className="search-form-input" />
      </form>
      <form className="filter-form" onChange={filterFormHandler}>
        <fieldset className="filter-form-fieldset">
          <h2 className="filter-form-heading">Media:</h2>
          {renderInputs({
            type: 'radio',
            category: 'media',
            labels: ['All', 'Film', 'Film (Episodes Only)', 'TV Series'],
            defaultChecked: 0
          })}
        </fieldset>
        <fieldset className="filter-form-fieldset">
          <h2 className="filter-form-heading">Era:</h2>
          {renderInputs({
            type: 'checkbox',
            category: 'era',
            labels: ['Prequel Trilogy', 'Original Trilogy', 'Sequel Trilogy']
          })}
        </fieldset>
        <fieldset className="filter-form-fieldset">
          <h2 className="filter-form-heading">Canon:</h2>
          <div className="filter-form-input-wrapper">
            <input
              type="checkbox"
              className="filter-form-input"
              id="my-canon"
              name="my-canon"
              value="my-canon"
            />
            <label className="filter-form-label" htmlFor="my-canon">
              My Canon
            </label>
          </div>
        </fieldset>
      </form>
    </div>
  );
}

FilterBar.propTypes = {
  filterFormHandler: PropTypes.func.isRequired,
  searchFormHandler: PropTypes.func.isRequired
};

export default FilterBar;
