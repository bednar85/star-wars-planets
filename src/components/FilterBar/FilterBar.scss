@import '../../styles/variables.scss';

.filter-bar {
  display: flex;
  flex-grow: 3;
  padding: 1.25rem;

  .search-form-input {
    background-image: url('../../assets/icons/search.svg');
    background-position: 0.625rem center;
    background-repeat: no-repeat;
    background-size: 8%;
    border-radius: 0.75rem;
    border: 2px solid #d8dde2;
    display: inline-block;
    font-family: 'Outage', sans-serif;
    font-size: 0.875rem;
    max-height: 3rem;
    outline: none;
    padding: 0.75rem;
    padding-left: 2.5rem;
    padding-right: 1rem;
    max-width: 15rem;
  }

  &__fieldset {
    border: none;
    flex-grow: 1;
    margin: 0;
    margin-left: 2.25rem;
    padding: 0;
  }

  &__heading {
    border-bottom: 2px solid $lightGray;
    font-family: 'Outage', sans-serif;
    font-size: 1rem;
    margin-bottom: 0.5rem;
    padding-bottom: 0.5rem;
  }

  &__input-wrapper {
    margin-bottom: 0.75rem;
    padding-left: 1.5rem;

    &--subset {
      padding-left: 3rem;
    }
  }

  &__input {
    display: none;
  }

  &__label {
    color: $lightGray;
    cursor: pointer;
    font-size: 0.75rem;

    .filter-bar__input:checked + & {
      color: $darkGray;
      position: relative;

      &:before {
        background-color: $lightGray;
        border-radius: 0.375rem;
        content: '';
        height: 0.75rem;
        left: -1.25rem;
        position: absolute;
        top: 50%;
        transform: translateY(-0.375rem);
        width: 0.75rem;
      }

      &[for='era-prequel']:before {
        background-color: $prequelColor;
      }

      &[for='era-original']:before {
        background-color: $originalColor;
      }

      &[for='era-sequel']:before {
        background-color: $sequelColor;
      }
    }
  }
}
