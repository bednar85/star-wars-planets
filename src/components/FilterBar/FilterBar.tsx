import React, { ReactElement } from 'react';
import { useForm } from 'react-hook-form';
import { FILTER_KEY } from '../../constants';

interface Input {
  type: string;
  name: string;
  labels: string[];
  defaultChecked?: number;
}

interface FilterBarProps {
  register: ReturnType<typeof useForm>['register'];
}

function FilterBar({ register }: FilterBarProps) {
  const renderInputs = ({
    type,
    name,
    labels,
    defaultChecked
  }: Input): ReactElement[] => {
    return labels.map((label: string, index: number) => {
      const isCheckedByDefault: boolean =
        defaultChecked !== undefined ? defaultChecked === index : false;
      const labelKebabCased: string = label
        .toLowerCase()
        .replace(/[^a-zA-Z']/gi, ' ')
        .trim()
        .replace(/\s+/g, '-');

      return (
        <div
          key={`filter-form-input-${index}`}
          className="filter-form-input-wrapper"
        >
          <input
            type={type}
            className="filter-form-input"
            id={`${name}-${labelKebabCased}`}
            name={name}
            value={label}
            defaultChecked={isCheckedByDefault}
            ref={register}
          />
          <label
            className="filter-form-label"
            htmlFor={`${name}-${labelKebabCased}`}
          >
            {label}
          </label>
        </div>
      );
    });
  };

  return (
    <form className="filter-form">
      <input
        type="text"
        name="searchQuery"
        className="search-form-input"
        ref={register}
      />
      <fieldset className="filter-form-fieldset">
        <h2 className="filter-form-heading">
          Media <em>(choose one)</em>
        </h2>
        {renderInputs({
          type: 'radio',
          name: FILTER_KEY.MEDIA,
          labels: ['All', 'Film', 'Film (Episodes Only)', 'TV Series'],
          defaultChecked: 0
        })}
      </fieldset>
      <fieldset className="filter-form-fieldset">
        <h2 className="filter-form-heading">Era</h2>
        {renderInputs({
          type: 'checkbox',
          name: FILTER_KEY.ERA,
          labels: ['Prequel Trilogy', 'Original Trilogy', 'Sequel Trilogy']
        })}
      </fieldset>
      <fieldset className="filter-form-fieldset">
        <h2 className="filter-form-heading">Canon</h2>
        <div className="filter-form-input-wrapper">
          <input
            type="checkbox"
            className="filter-form-input"
            id={FILTER_KEY.MY_CANON}
            name={FILTER_KEY.MY_CANON}
            ref={register}
          />
          <label className="filter-form-label" htmlFor={FILTER_KEY.MY_CANON}>
            My Canon
          </label>
        </div>
      </fieldset>
    </form>
  );
}

export default FilterBar;
