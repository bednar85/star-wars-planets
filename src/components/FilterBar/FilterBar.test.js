import FilterBar from './FilterBar';

describe('App Component', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <FilterBar searchFormHandler={() => {}} filterFormHandler={() => {}} />
    );

    expect(wrapper).toMatchSnapshot();
  });
});
