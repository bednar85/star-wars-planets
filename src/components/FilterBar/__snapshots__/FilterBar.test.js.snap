// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <form
        class="filter-form"
      >
        <input
          class="search-form-input"
          name="searchQuery"
          type="text"
        />
        <fieldset
          class="filter-form__fieldset"
        >
          <h2
            class="filter-form__heading"
          >
            Media
          </h2>
          <div
            class="filter-form__input-wrapper "
          >
            <input
              class="filter-form__input"
              id="media-all"
              name="media"
              type="radio"
              value="All"
            />
            <label
              class="filter-form__label"
              for="media-all"
            >
              All â€“ ðŸŽ¥ ðŸ“º
            </label>
          </div>
          <div
            class="filter-form__input-wrapper "
          >
            <input
              class="filter-form__input"
              id="media-film"
              name="media"
              type="radio"
              value="Film"
            />
            <label
              class="filter-form__label"
              for="media-film"
            >
              Film â€“ ðŸŽ¥
            </label>
          </div>
          <div
            class="filter-form__input-wrapper filter-form__input-wrapper--subset"
          >
            <input
              class="filter-form__input"
              id="media-episodes"
              name="media"
              type="radio"
              value="Episodes"
            />
            <label
              class="filter-form__label"
              for="media-episodes"
            >
              Episodes
            </label>
          </div>
          <div
            class="filter-form__input-wrapper filter-form__input-wrapper--subset"
          >
            <input
              class="filter-form__input"
              id="media-spinoffs"
              name="media"
              type="radio"
              value="Spinoffs"
            />
            <label
              class="filter-form__label"
              for="media-spinoffs"
            >
              Spinoffs
            </label>
          </div>
          <div
            class="filter-form__input-wrapper "
          >
            <input
              class="filter-form__input"
              id="media-tv-series"
              name="media"
              type="radio"
              value="TV Series"
            />
            <label
              class="filter-form__label"
              for="media-tv-series"
            >
              TV Series â€“ ðŸ“º
            </label>
          </div>
        </fieldset>
        <fieldset
          class="filter-form__fieldset"
        >
          <h2
            class="filter-form__heading"
          >
            Era
          </h2>
          <div
            class="filter-form__input-wrapper "
          >
            <input
              class="filter-form__input"
              id="era-prequel"
              name="era"
              type="checkbox"
              value="Prequel"
            />
            <label
              class="filter-form__label"
              for="era-prequel"
            >
              Prequel
            </label>
          </div>
          <div
            class="filter-form__input-wrapper "
          >
            <input
              class="filter-form__input"
              id="era-original"
              name="era"
              type="checkbox"
              value="Original"
            />
            <label
              class="filter-form__label"
              for="era-original"
            >
              Original
            </label>
          </div>
          <div
            class="filter-form__input-wrapper "
          >
            <input
              class="filter-form__input"
              id="era-sequel"
              name="era"
              type="checkbox"
              value="Sequel"
            />
            <label
              class="filter-form__label"
              for="era-sequel"
            >
              Sequel
            </label>
          </div>
        </fieldset>
        <fieldset
          class="filter-form__fieldset"
        >
          <h2
            class="filter-form__heading"
          >
            Canon
          </h2>
          <div
            class="filter-form__input-wrapper"
          >
            <input
              class="filter-form__input"
              id="myCanon"
              name="myCanon"
              type="checkbox"
            />
            <label
              class="filter-form__label"
              for="myCanon"
            >
              My Canon
            </label>
          </div>
        </fieldset>
      </form>
    </div>
  </body>,
  "container": <div>
    <form
      class="filter-form"
    >
      <input
        class="search-form-input"
        name="searchQuery"
        type="text"
      />
      <fieldset
        class="filter-form__fieldset"
      >
        <h2
          class="filter-form__heading"
        >
          Media
        </h2>
        <div
          class="filter-form__input-wrapper "
        >
          <input
            class="filter-form__input"
            id="media-all"
            name="media"
            type="radio"
            value="All"
          />
          <label
            class="filter-form__label"
            for="media-all"
          >
            All â€“ ðŸŽ¥ ðŸ“º
          </label>
        </div>
        <div
          class="filter-form__input-wrapper "
        >
          <input
            class="filter-form__input"
            id="media-film"
            name="media"
            type="radio"
            value="Film"
          />
          <label
            class="filter-form__label"
            for="media-film"
          >
            Film â€“ ðŸŽ¥
          </label>
        </div>
        <div
          class="filter-form__input-wrapper filter-form__input-wrapper--subset"
        >
          <input
            class="filter-form__input"
            id="media-episodes"
            name="media"
            type="radio"
            value="Episodes"
          />
          <label
            class="filter-form__label"
            for="media-episodes"
          >
            Episodes
          </label>
        </div>
        <div
          class="filter-form__input-wrapper filter-form__input-wrapper--subset"
        >
          <input
            class="filter-form__input"
            id="media-spinoffs"
            name="media"
            type="radio"
            value="Spinoffs"
          />
          <label
            class="filter-form__label"
            for="media-spinoffs"
          >
            Spinoffs
          </label>
        </div>
        <div
          class="filter-form__input-wrapper "
        >
          <input
            class="filter-form__input"
            id="media-tv-series"
            name="media"
            type="radio"
            value="TV Series"
          />
          <label
            class="filter-form__label"
            for="media-tv-series"
          >
            TV Series â€“ ðŸ“º
          </label>
        </div>
      </fieldset>
      <fieldset
        class="filter-form__fieldset"
      >
        <h2
          class="filter-form__heading"
        >
          Era
        </h2>
        <div
          class="filter-form__input-wrapper "
        >
          <input
            class="filter-form__input"
            id="era-prequel"
            name="era"
            type="checkbox"
            value="Prequel"
          />
          <label
            class="filter-form__label"
            for="era-prequel"
          >
            Prequel
          </label>
        </div>
        <div
          class="filter-form__input-wrapper "
        >
          <input
            class="filter-form__input"
            id="era-original"
            name="era"
            type="checkbox"
            value="Original"
          />
          <label
            class="filter-form__label"
            for="era-original"
          >
            Original
          </label>
        </div>
        <div
          class="filter-form__input-wrapper "
        >
          <input
            class="filter-form__input"
            id="era-sequel"
            name="era"
            type="checkbox"
            value="Sequel"
          />
          <label
            class="filter-form__label"
            for="era-sequel"
          >
            Sequel
          </label>
        </div>
      </fieldset>
      <fieldset
        class="filter-form__fieldset"
      >
        <h2
          class="filter-form__heading"
        >
          Canon
        </h2>
        <div
          class="filter-form__input-wrapper"
        >
          <input
            class="filter-form__input"
            id="myCanon"
            name="myCanon"
            type="checkbox"
          />
          <label
            class="filter-form__label"
            for="myCanon"
          >
            My Canon
          </label>
        </div>
      </fieldset>
    </form>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
